string space_name
uint16 traj_cnt